### ISSUE TITLE
Initialization failure due to corrupted /data/local/tmp

### ISSUE THREAD

**Issue by Cyberavater** [MuntashirAkon/AppManager#534]
I've opened a new issue since it's a different issue altogether. (from #473)

A:10 (LOS based), Rooted.

>Send me some logs during the initialisation. They must be unfiltered.

Log:

```
2021-07-26 14:02:11.195 W/KSManager: Using keystore BKS
2021-07-26 14:02:11.215 I/CompatUtil: Loading local protection key
2021-07-26 14:02:11.225 I/CompatUtil: Generating AES key with keystore
2021-07-26 14:02:11.249 E/KS: Could not get encrypted password
java.security.ProviderException: Keystore operation failed
	at android.security.keystore.AndroidKeyStoreKeyGeneratorSpi.engineGenerateKey(AndroidKeyStoreKeyGeneratorSpi.java:332)
	at javax.crypto.KeyGenerator.generateKey(KeyGenerator.java:612)
	at io.github.muntashirakon.AppManager.crypto.ks.CompatUtil.getAesGcmLocalProtectionKey(CompatUtil.java:116)
	at io.github.muntashirakon.AppManager.crypto.ks.CompatUtil.getEncryptedData(CompatUtil.java:215)
	at io.github.muntashirakon.AppManager.crypto.ks.KeyStoreManager.getEncryptedPassword(KeyStoreManager.java:430)
	at io.github.muntashirakon.AppManager.crypto.ks.KeyStoreManager.savePass(KeyStoreManager.java:399)
	at io.github.muntashirakon.AppManager.crypto.ks.KeyStoreManager.generateAndDisplayKeyStorePassword(KeyStoreManager.java:108)
	at io.github.muntashirakon.AppManager.crypto.AuthenticationActivity.onCreate(AuthenticationActivity.java:68)
	at android.app.Activity.performCreate(Activity.java:7824)
	at android.app.Activity.performCreate(Activity.java:7813)
	at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1307)
	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:3245)
	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:3409)
	at android.app.servertransaction.LaunchActivityItem.execute(LaunchActivityItem.java:83)
	at android.app.servertransaction.TransactionExecutor.executeCallbacks(TransactionExecutor.java:135)
	at android.app.servertransaction.TransactionExecutor.execute(TransactionExecutor.java:95)
	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:2016)
	at android.os.Handler.dispatchMessage(Handler.java:107)
	at android.os.Looper.loop(Looper.java:214)
	at android.app.ActivityThread.main(ActivityThread.java:7357)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:925)
Caused by: android.security.KeyStoreException: System error
	at android.security.KeyStore.getKeyStoreException(KeyStore.java:1279)
	... 23 more
2021-07-26 14:02:15.922 D/Runner: RootShellRunner
2021-07-26 14:02:16.121 D/IPCUtils: Launching service...
2021-07-26 14:02:16.130 D/IPC: Total bound services: 0
2021-07-26 14:02:16.131 D/IPC: Need to start a new connection.
2021-07-26 14:02:16.141 E/IPC: Running service starter script...
2021-07-26 14:03:01.128 E/ROOT
android.os.RemoteException: AMService not running.
	at io.github.muntashirakon.AppManager.ipc.IPCUtils$AMServiceConnectionWrapper.getServiceSafe(IPCUtils.java:185)
	at io.github.muntashirakon.AppManager.ipc.IPCUtils$AMServiceConnectionWrapper.getAmService(IPCUtils.java:171)
	at io.github.muntashirakon.AppManager.ipc.IPCUtils.getAmService(IPCUtils.java:44)
	at io.github.muntashirakon.AppManager.servermanager.LocalServer.launchAmService(LocalServer.java:60)
	at io.github.muntashirakon.AppManager.runner.RunnerUtils.autoDetectRootOrAdb(RunnerUtils.java:166)
	at io.github.muntashirakon.AppManager.runner.RunnerUtils.setModeOfOps(RunnerUtils.java:190)
	at io.github.muntashirakon.AppManager.crypto.AuthenticationActivity.lambda$handleModeOfOps$2$AuthenticationActivity(AuthenticationActivity.java:107)
	at io.github.muntashirakon.AppManager.crypto.-$$Lambda$AuthenticationActivity$RV-B2vNmyKZyexsvVCuRvnoLqh4.run(Unknown Source:2)
	at java.lang.Thread.run(Thread.java:919)
2021-07-26 14:03:02.399 D/MVM: New instance created
2021-07-26 14:03:04.414 E/MetadataManager: java.io.FileNotFoundException: Cannot find meta_v2.am.json
2021-07-26 14:03:05.396 E/Users: Could not get list of users
java.lang.SecurityException: You either need MANAGE_USERS or CREATE_USERS permission to: query users
	at android.os.Parcel.createException(Parcel.java:2071)
	at android.os.Parcel.readException(Parcel.java:2039)
	at android.os.Parcel.readException(Parcel.java:1987)
	at android.os.IUserManager$Stub$Proxy.getUsers(IUserManager.java:1747)
	at io.github.muntashirakon.AppManager.users.Users.getAllUsers(Users.java:72)
	at io.github.muntashirakon.AppManager.users.Users.getUsersIds(Users.java:101)
	at io.github.muntashirakon.AppManager.utils.PackageUtils.updateInstalledOrBackedUpApplications(PackageUtils.java:267)
	at io.github.muntashirakon.AppManager.utils.PackageUtils.lambda$getInstalledOrBackedUpApplicationsFromDb$0(PackageUtils.java:173)
	at io.github.muntashirakon.AppManager.utils.-$$Lambda$PackageUtils$lQaPt-sS9Llm_I9c5WeeA5MEQDs.run(Unknown Source:4)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:462)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1167)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:641)
	at java.lang.Thread.run(Thread.java:919)
Caused by: android.os.RemoteException: Remote stack trace:
	at com.android.server.pm.UserManagerService.checkManageOrCreateUsersPermission(UserManagerService.java:1988)
	at com.android.server.pm.UserManagerService.getUsers(UserManagerService.java:695)
	at android.os.IUserManager$Stub.onTransact(IUserManager.java:785)
	at android.os.Binder.execTransactInternal(Binder.java:1021)
	at android.os.Binder.execTransact(Binder.java:994)
	
```

>What do you mean?

![image](https://user-images.githubusercontent.com/32187927/127427567-5fffecca-e57f-4a8f-a19d-ad26fee92317.png)

**Comment by MuntashirAkon** [comment]
The logs are not unfiltered. For instance, it didn't list anything between `E/IPC: Running service starter script...` and `E/ROOT` which is/are the most important logs. These logs are recorded in another process, outside of App Manager, so they aren't logged anywhere except in logcat. But try 48f7ae65 before that.

> ![image](https://user-images.githubusercontent.com/32187927/127427567-5fffecca-e57f-4a8f-a19d-ad26fee92317.png)

Weird! It should've at least find `/storage/emulated/0`. Set AM to no-root mode and check again.

**Comment by Cyberavater** [comment]
So, I should collect logs with another app?
Also, which version I should test? 2.6 or 2.7? The last one in TG is 2.6

**Comment by MuntashirAkon** [comment]
> So, I should collect logs with another app?

You can still take logs in AM if `android.permission.DUMP` is granted to it. Otherwise, use alternatives such as Matlog.

> Also, which version I should test? 2.6 or 2.7? The last one in TG is 2.6

v2.7.0, of course. Commits are only backported to v2.6.x. Latest debug version is always v2.7.0.

**Comment by Cyberavater** [comment]
Log:

```
44:40.638 I/Finsky  (3943): [450] afnx.c(3): Verify: Verification package=io.github.muntashirakon.AppManager.debug, id=1 response=0
44:40.806 I/ActivityManager(1393): Force stopping io.github.muntashirakon.AppManager.debug appid=10295 user=-1: installPackageLI
44:40.822 I/PackageManager(1393): Update package io.github.muntashirakon.AppManager.debug code path from /data/app/io.github.muntashirakon.AppManager.debug-5qcOUzJ0zDXNKswUm8IEeQ== to /data/app/io.github.muntashirakon.AppManager.debug-8dzUCQCC-_sKQ0wK34RFvw==; Retain data and using new
44:40.822 I/PackageManager(1393): Update package io.github.muntashirakon.AppManager.debug resource path from /data/app/io.github.muntashirakon.AppManager.debug-5qcOUzJ0zDXNKswUm8IEeQ== to /data/app/io.github.muntashirakon.AppManager.debug-8dzUCQCC-_sKQ0wK34RFvw==; Retain data and using new
44:41.604 I/PackageManager(1393): Un-granting permission android.permission.DUMP from package io.github.muntashirakon.AppManager.debug (protectionLevel=50 flags=0x30583e46)
44:41.604 I/PackageManager(1393): Un-granting permission android.permission.GET_APP_OPS_STATS from package io.github.muntashirakon.AppManager.debug (protectionLevel=50 flags=0x30583e46)
44:41.604 I/PackageManager(1393): Un-granting permission android.permission.MANAGE_USERS from package io.github.muntashirakon.AppManager.debug (protectionLevel=18 flags=0x30583e46)
44:41.604 I/PackageManager(1393): Un-granting permission android.permission.PACKAGE_USAGE_STATS from package io.github.muntashirakon.AppManager.debug (protectionLevel=114 flags=0x30583e46)
44:41.604 I/PackageManager(1393): Un-granting permission android.permission.READ_LOGS from package io.github.muntashirakon.AppManager.debug (protectionLevel=50 flags=0x30583e46)
44:41.604 I/PackageManager(1393): Un-granting permission android.permission.REQUEST_INSTALL_PACKAGES from package io.github.muntashirakon.AppManager.debug (protectionLevel=66 flags=0x30583e46)
44:42.105 W/cutils  (955): Expected path /data/user_de/0/io.github.muntashirakon.AppManager.debug with mode 700 but found 711
44:42.139 E/cutils  (955): Failed to open(/data/misc/profiles/cur/0/io.github.muntashirakon.AppManager.debug/primary.prof): No such file or directory
44:42.140 E/installd(955): Failed to prepare /data/misc/profiles/cur/0/io.github.muntashirakon.AppManager.debug/primary.prof: No such file or directory
44:42.140 E/ArtManagerService(1393): Failed to prepare profile for io.github.muntashirakon.AppManager.debug:/data/app/io.github.muntashirakon.AppManager.debug-8dzUCQCC-_sKQ0wK34RFvw==/base.apk
44:42.164 I/ActivityManager(1393): Force stopping io.github.muntashirakon.AppManager.debug appid=10295 user=0: pkg removed
44:42.194 I/ActivityManager(1393): Force stopping io.github.muntashirakon.AppManager.debug appid=10295 user=0: pkg removed
44:42.215 I/Finsky  (3943): [2] tsn.n(8): Invalidating cached PackageState for io.github.muntashirakon.AppManager.debug
44:42.216 I/GsaVoiceInteractionSrv(2707): O received Intent { act=android.intent.action.PACKAGE_REMOVED dat=package:io.github.muntashirakon.AppManager.debug flg=0x4000010 (has extras) }
44:42.217 D/IconPackManager(2085): Received intent action android.intent.action.PACKAGE_ADDED for io.github.muntashirakon.AppManager.debug
44:42.248 W/BroadcastQueue(1393): Background execution not allowed: receiving Intent { act=android.intent.action.PACKAGE_REMOVED dat=package:io.github.muntashirakon.AppManager.debug flg=0x4000010 (has extras) } to com.mixplorer/.BroadcastReceiver
44:42.266 I/GsaVoiceInteractionSrv(2707): O received Intent { act=android.intent.action.PACKAGE_ADDED dat=package:io.github.muntashirakon.AppManager.debug flg=0x4000010 (has extras) }
...
44:43.109 W/BroadcastQueue(1393): Background execution not allowed: receiving Intent { act=android.intent.action.PACKAGE_REPLACED dat=package:io.github.muntashirakon.AppManager.debug flg=0x4000010 (has extras) } to com.google.android.apps.photos/.notificationchannels.AppUpdateBroadcastReceiver
44:43.110 W/BroadcastQueue(1393): Background execution not allowed: receiving Intent { act=android.intent.action.PACKAGE_REPLACED dat=package:io.github.muntashirakon.AppManager.debug flg=0x4000010 (has extras) } to com.topjohnwu.magisk/.core.Receiver
44:44.152 I/ActivityTaskManager(1393): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10000000 pkg=io.github.muntashirakon.AppManager.debug cmp=io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.main.MainActivity} from uid 10151
44:44.242 I/ActivityManager(1393): Start proc 5651:io.github.muntashirakon.AppManager.debug/u0a295 for pre-top-activity {io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.main.MainActivity}
44:44.736 W/SQLiteLog(2544): (28) double-quoted string literal: "io.github.muntashirakon.AppManager.debug"
44:46.480 I/ActivityTaskManager(1393): START u0 {cmp=io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.crypto.AuthenticationActivity} from uid 10295
44:46.570 V/AppLockService(1393): setForegroundApp(io.github.muntashirakon.AppManager.debug)
44:47.945 V/AppLockService(1393): setForegroundApp(io.github.muntashirakon.AppManager.debug)
44:48.767 W/nce     (2315): Infinite Data has no result for 9677661882425161_io.github.muntashirakon.AppManager.debug [CONTEXT service_id=177 ]
44:48.839 W/nce     (2315): Infinite Data has no result for 9677661882425161_io.github.muntashirakon.AppManager.debug [CONTEXT service_id=177 ]
44:49.089 I/DeviceAppInfoWork(12260): remove existing package for update: io.github.muntashirakon.AppManager.debug
44:49.093 I/DeviceAppInfoWork(12260): remove existing package for update: io.github.muntashirakon.AppManager.debug
44:49.099 I/DeviceAppInfoWork(12260): remove existing package for update: io.github.muntashirakon.AppManager.debug
44:49.848 E/Layer   (907): [Surface(name=AppWindowToken{4faeb3a token=Token{24eb265 ActivityRecord{443bf5c u0 io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.main.MainActivity t469}}})/@0xed1d24a - animation-leash#0] No local sync point found
44:49.850 E/Layer   (907): [Surface(name=AppWindowToken{4faeb3a token=Token{24eb265 ActivityRecord{443bf5c u0 io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.main.MainActivity t469}}})/@0xed1d24a - animation-leash#0] No local sync point found
44:49.850 E/Layer   (907): [Surface(name=AppWindowToken{1381f98 token=Token{174a57b ActivityRecord{347d90a u0 io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.crypto.AuthenticationActivity t469}}})/@0xed241cc - animation-leash#0] No local sync point found
44:49.850 E/Layer   (907): [Surface(name=AppWindowToken{1381f98 token=Token{174a57b ActivityRecord{347d90a u0 io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.crypto.AuthenticationActivity t469}}})/@0xed241cc - animation-leash#0] No local sync point found
44:49.875 E/Layer   (907): [Surface(name=AppWindowToken{4faeb3a token=Token{24eb265 ActivityRecord{443bf5c u0 io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.main.MainActivity t469}}})/@0xed1d24a - animation-leash#0] No local sync point found
44:49.875 E/Layer   (907): [Surface(name=AppWindowToken{1381f98 token=Token{174a57b ActivityRecord{347d90a u0 io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.crypto.AuthenticationActivity t469}}})/@0xed241cc - animation-leash#0] No local sync point found
44:50.730 I/ActivityManager(1393): Killing 5651:io.github.muntashirakon.AppManager.debug/u0a295 (adj 905): remove task
44:50.807 D/AutofillSession(1393): handling death of Token{174a57b ActivityRecord{347d90a u0 io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.crypto.AuthenticationActivity t469 f}} when saving=false
44:50.808 I/WindowManager(1393): WIN DEATH: Window{71749b0 u0 io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.main.MainActivity}
44:50.812 I/WindowManager(1393): WIN DEATH: Window{c983ed1 u0 io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.crypto.AuthenticationActivity}
44:50.816 I/WindowManager(1393): WIN DEATH: Window{620a46a u0 io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.crypto.AuthenticationActivity}
45:00.237 I/ActivityTaskManager(1393): START u0 {act=android.settings.APPLICATION_DETAILS_SETTINGS dat=package:io.github.muntashirakon.AppManager.debug flg=0x10008000 cmp=com.android.settings/.applications.InstalledAppDetails bnds=[75,798][735,966]} from uid 10894
45:02.218 W/Role    (18005): io.github.muntashirakon.AppManager.debug not qualified for android.app.role.HOME due to missing RequiredComponent{mIntentFilterData=IntentFilterData{mAction='android.intent.action.MAIN', mCategories='[android.intent.category.HOME]', mDataScheme='null', mDataType='null'}, mPermission='null', mMetaData=[]}
45:02.233 W/Role    (18005): io.github.muntashirakon.AppManager.debug not qualified for android.app.role.DIALER due to missing RequiredComponent{mIntentFilterData=IntentFilterData{mAction='android.intent.action.DIAL', mCategories='[]', mDataScheme='null', mDataType='null'}, mPermission='null', mMetaData=[]}
45:02.242 W/Role    (18005): io.github.muntashirakon.AppManager.debug not qualified for android.app.role.SMS due to missing RequiredComponent{mIntentFilterData=IntentFilterData{mAction='android.provider.Telephony.SMS_DELIVER', mCategories='[]', mDataScheme='null', mDataType='null'}, mPermission='android.permission.BROADCAST_SMS', mMetaData=[]}
45:06.526 I/ActivityManager(1393): Force stopping io.github.muntashirakon.AppManager.debug appid=10295 user=0: from pid 15223
45:11.928 I/ActivityManager(1393): Force stopping io.github.muntashirakon.AppManager.debug appid=10295 user=0: clear data
45:11.939 I/ActivityManager(1393): Force stopping io.github.muntashirakon.AppManager.debug appid=10295 user=-1: clearApplicationUserData
45:11.948 I/ConditionProviders(1393): Disallowing condition provider io.github.muntashirakon.AppManager.debug
45:12.571 I/MediaProvider(15555): Begin Intent { act=android.intent.action.PACKAGE_DATA_CLEARED dat=package:io.github.muntashirakon.AppManager.debug flg=0x1000010 cmp=com.android.providers.media/.MediaService (has extras) }
45:12.626 I/BlockstoreStorage(2544): Writing 0 bytes of block data for io.github.muntashirakon.AppManager.debug [CONTEXT service_id=258 ]
45:12.626 I/Icing   (2544): doRemovePackageData io.github.muntashirakon.AppManager.debug
45:13.150 I/MediaProvider(15555): End Intent { act=android.intent.action.PACKAGE_DATA_CLEARED dat=package:io.github.muntashirakon.AppManager.debug flg=0x1000010 cmp=com.android.providers.media/.MediaService (has extras) }
45:13.498 D/AppButtonsPrefCtl(15223): Sending broadcast to query restart status for io.github.muntashirakon.AppManager.debug
45:14.090 D/AppButtonsPrefCtl(15223): Got broadcast response: Restart status for io.github.muntashirakon.AppManager.debug false
45:14.144 D/AppButtonsPrefCtl(15223): Got broadcast response: Restart status for io.github.muntashirakon.AppManager.debug false
45:56.364 I/ActivityTaskManager(1393): START u0 {act=android.intent.action.MAIN cat=[android.intent.category.LAUNCHER] flg=0x10200000 cmp=io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.main.MainActivity bnds=[123,574][303,754]} from uid 10894
45:56.456 I/ActivityManager(1393): Start proc 15226:io.github.muntashirakon.AppManager.debug/u0a295 for pre-top-activity {io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.main.MainActivity}
45:57.835 I/ActivityTaskManager(1393): START u0 {cmp=io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.crypto.AuthenticationActivity} from uid 10295
45:57.917 V/AppLockService(1393): setForegroundApp(io.github.muntashirakon.AppManager.debug)
45:58.030 I/ActivityTaskManager(1393): Displayed io.github.muntashirakon.AppManager.debug/io.github.muntashirakon.AppManager.crypto.AuthenticationActivity: +1s665ms
45:58.819 V/AppLockService(1393): setForegroundApp(io.github.muntashirakon.AppManager.debug)
45:59.424 W/nce     (2315): Infinite Data has no result for 9677661882425161_io.github.muntashirakon.AppManager.debug [CONTEXT service_id=177 ]
45:59.432 W/nce     (2315): Infinite Data has no result for 9677661882425161_io.github.muntashirakon.AppManager.debug [CONTEXT service_id=177 ]
46:07.128 I/ActivityTaskManager(1393): START u0 {act=android.settings.APPLICATION_DETAILS_SETTINGS dat=package:io.github.muntashirakon.AppManager.debug flg=0x10008000 cmp=com.android.settings/.applications.InstalledAppDetails bnds=[333,1113][993,1281]} from uid 10894
46:07.325 D/AppButtonsPrefCtl(15223): Sending broadcast to query restart status for io.github.muntashirakon.AppManager.debug
46:08.075 D/AppButtonsPrefCtl(15223): Got broadcast response: Restart status for io.github.muntashirakon.AppManager.debug true
46:08.152 W/Role    (18005): io.github.muntashirakon.AppManager.debug not qualified for android.app.role.HOME due to missing RequiredComponent{mIntentFilterData=IntentFilterData{mAction='android.intent.action.MAIN', mCategories='[android.intent.category.HOME]', mDataScheme='null', mDataType='null'}, mPermission='null', mMetaData=[]}
46:08.166 W/Role    (18005): io.github.muntashirakon.AppManager.debug not qualified for android.app.role.DIALER due to missing RequiredComponent{mIntentFilterData=IntentFilterData{mAction='android.intent.action.DIAL', mCategories='[]', mDataScheme='null', mDataType='null'}, mPermission='null', mMetaData=[]}
46:08.177 W/Role    (18005): io.github.muntashirakon.AppManager.debug not qualified for android.app.role.SMS due to missing RequiredComponent{mIntentFilterData=IntentFilterData{mAction='android.provider.Telephony.SMS_DELIVER', mCategories='[]', mDataScheme='null', mDataType='null'}, mPermission='android.permission.BROADCAST_SMS', mMetaData=[]}

```

**Comment by MuntashirAkon** [comment]
Logs are still filtered using package name. Perhaps you aren't aware: filtering logs by package name doesn't actually work. For instance, it missed all the previous logs. If you're concerned with privacy, send me the logs via email or TG.

**Comment by Cyberavater** [comment]
I've sent the logs using TG if you still hold the same account.

**Comment by MuntashirAkon** [comment]
The reason behind this issue was `/data/local/tmp` not being present in the system (or existed as a file). The solution is to check the existence of the directory and recreate if it does not exist.

According to [this issue](https://bugzilla.mozilla.org/show_bug.cgi?id=879489), this folder may have the wrong permissions or may exist as a file depending on device.

**Comment by MuntashirAkon** [comment]
This also means that we've to check this folder during APK installation for root/ADB users since installed APK is kept there temporarily.

**Comment by MuntashirAkon** [comment]
Fixed in 06f31b8f
